packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -649,6 +673,70 @@ contract Attestations is
     }
   }
 
+  /**
+   * @notice Update the approval status of allowing an attestation identifier
+   * mapping to be transfered from an address to another.  The "to" or "from"
+   * addresses must both approve.  If the other has already approved, then the transfer
+   * is executed.
+   * @param identifier The identifier for this attestation.
+   * @param from The current attestation address to which the identifier is mapped.
+   * @param to The new address to map to identifier.
+   * @param status The approval status
+   */
+  function approveTransfer(bytes32 identifier, uint256 index, address from, address to, bool status)
+    external
+  {
+    require(from != address(0) && to != address(0), "reserved address 0x0 cannot have approval");
+    bytes32 key = keccak256(abi.encodePacked(identifier, from, to));
```

Comment:
Minor suggestion that would allow you to remove `isOtherAddressApproved`
```
    require(msg.sender == from || msg.sender == to);
    bytes32 key = keccak256(abi.encodePacked(identifier, from, to));
    address other = msg.sender == from ? to : from;
    if (status && transferApprovals[other][key]) {
      transferApprovals[other][key] = false;
      _transfer(identifier, index, from, to);
    } else {
      transferApprovals[msg.sender][key] = status;
      emit TransferApproval(msg.sender, from, to, status);
    }
```

Reply:
ya -- that's a good way to do it.  i'll update

Reply:
I don't think this code works without the first `require`